lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,117 @@
+import 'dart:collection';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// Name of the license, is empty in case of unknown license.
+  final String licenseName;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// A map of tokens and their count.
+  final Map<String, int> occurences;
+
+  License._(this.content, this.tokens, this.occurences, this.licenseName);
+
+  factory License.parse(String licenseName, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, licenseName);
+  }
+}
+
+// Commented out as they are not yet needed in this stage of detection.
+// class Checksum {
+//   /// Text for which the hash value was generated.
+//   final String text;
+
+//   /// [Crc-32][1] checksum value generated for text.
+//   ///
+//   /// [1]: https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+//   final int crc32;
+
+//   Checksum(this.text, this.crc32);
+// }
+
+class Result {
```

Comment:
Just suggesting a better name, I don't know if it is actually better... I was thinking that a `RegExp` returns a `Match` object, so why shouldn't we return a `LicenseMatch` object? hehe

```suggestion
@sealed
class LicenseMatch {
```

Feel free to ignore :D